ktor-features/ktor-serialization/jvm/test/io/ktor/tests/serialization/SerializationTest.kt

```diff
@@ -416,6 +417,50 @@ class SerializationTest {
             assertEquals("""{"a":"1",null:"2","b":null}""", it.response.content)
         }
     }
+
+    @Test
+    fun testRespondPolymorphic(): Unit = withTestApplication {
+        application.install(ContentNegotiation) {
+            register(ContentType.Application.Json, SerializationConverter(Json))
+        }
+        application.routing {
+            get("/sealed") {
+                call.respond(listOf(TestSealed.A("valueA"), TestSealed.B("valueB")))
+            }
+        }
+
+        handleRequest(HttpMethod.Get, "/sealed") {
+            addHeader("Accept", "application/json")
+        }.let { call ->
+            assertEquals(HttpStatusCode.OK, call.response.status())
+            assertEquals(
+                """[{"type":"io.ktor.tests.serialization.TestSealed.A","valueA":"valueA"},{"type":"io.ktor.tests.serialization.TestSealed.B","valueB":"valueB"}]""",
+                call.response.content
+            )
+        }
+    }
+
+    @Test
+    fun testRespondAny(): Unit = withTestApplication {
+        application.install(ContentNegotiation) {
+            register(ContentType.Application.Json, SerializationConverter(Json))
+        }
+        application.routing {
+            get("/") {
+                call.respond(listOf(TextPlainData(777), TextPlainData(888)) as Any)
+            }
+        }
+
+        handleRequest(HttpMethod.Get, "/") {
+            addHeader("Accept", "application/json")
+        }.let { call ->
+            assertEquals(HttpStatusCode.OK, call.response.status())
+            assertEquals(
+                """[{"x":777},{"x":888}]""",
+                call.response.content
+            )
+        }
+    }
```

Comment:
I would also test `null` (both typed `null` and `Nothing?`), and a case with `respond(when { ...})`

Reply:
added test for response `when`